pkg/kubelet/cm/topologymanager/policy_single_numa_node.go

```diff
@@ -17,27 +17,33 @@ limitations under the License.
 package topologymanager
 
 import (
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/bitmask"
 	"k8s.io/kubernetes/pkg/kubelet/lifecycle"
+	"sort"
 )
 
-type singleNumaNodePolicy struct{}
+type singleNumaNodePolicy struct {
+	//List of NUMA Nodes available on the underlying machine
+	numaNodes []int
+}
 
 var _ Policy = &singleNumaNodePolicy{}
 
 // PolicySingleNumaNode policy name.
 const PolicySingleNumaNode string = "single-numa-node"
 
 // NewSingleNumaNodePolicy returns single-numa-node policy.
-func NewSingleNumaNodePolicy() Policy {
-	return &singleNumaNodePolicy{}
+func NewSingleNumaNodePolicy(numaNodes []int) Policy {
+	return &singleNumaNodePolicy{numaNodes: numaNodes}
 }
 
 func (p *singleNumaNodePolicy) Name() string {
 	return PolicySingleNumaNode
 }
 
-func (p *singleNumaNodePolicy) CanAdmitPodResult(hint *TopologyHint) lifecycle.PodAdmitResult {
-	if !hint.Preferred || hint.NUMANodeAffinity.Count() > 1 {
+func (p *singleNumaNodePolicy) canAdmitPodResult(hint *TopologyHint) lifecycle.PodAdmitResult {
+	if !hint.Preferred || hint.NUMANodeAffinity.Count() != 1 {
```

Comment:
Do we have to consider whether the final hint is preferred or not?
I don't know how the device-plugin generates hint and set the preferred flag.

Reply:
since the logic is now separate, it is possible rely on either IMO.

please note that the changes (apart of the merge logic change) is already merged in PR #84721

This PR needs to be modified to propose the merge logic change only.
(or alternatively kill this PR and propose a new one)